recipes/publish.rb

```diff
@@ -38,41 +29,41 @@
 artifact_pkgident = nil
 last_build_env = nil
 
-ruby_block 'build-plan' do
-  block do
-    Dir.chdir(node['delivery']['workspace']['repo'])
-    ENV['TERM'] = 'ansi'
-    command = "sudo #{::File.join('/hab/pkgs', hab_studio_pkgident, 'bin/hab-studio')}"
-    command << " -r /hab/studios/#{studio_slug}"
-    command << " build #{habitat_plan_dir}"
-
-    build = shell_out(command)
-
-    if build.exitstatus > 0
-      puts build.stdout
-      puts build.stderr
-      raise 'The plan.sh did NOT come together, bailing out!'
-    end
+execute 'build-plan' do
+  command <<-EOH
+sudo #{hab_studio_binary} \
+-r #{hab_studio_path} \
+-k #{keyname.split('-')[0...-1].join('-')} \
+build #{habitat_plan_dir}
+  EOH
+  env(
+    'TERM' => 'ansi'
+  )
+  cwd node['delivery']['workspace']['repo']
+  live_stream true
+end
 
-    last_build_env = Hash[*::File.read(::File.join('/hab/studios',
-                                                   studio_slug,
+ruby_block 'load-build-output' do
+  block do
+    last_build_env = Hash[*::File.read(::File.join(hab_studio_path,
                                                    'src/results/last_build.env')).split(/[=\n]/)]
 
     artifact = last_build_env['pkg_artifact']
     artifact_pkgident = last_build_env['pkg_ident']
   end
 end
 
-ruby_block 'artifact-hash' do
+ruby_block 'generate-pkg-hash' do
   block do
-    command = "/hab/pkgs/#{hab_pkgident}/bin/hab"
-    command << " artifact hash #{::File.join(studio_path, '/src/results', artifact)}"
+    command = hab_binary
+    command << " pkg hash #{::File.join(hab_studio_path, '/src/results', artifact)}"
     artifact_hash = shell_out(command).stdout.chomp
   end
 end
 
-execute 'upload-artifact' do
-  command lazy { "#{::File.join('/hab/pkgs', hab_pkgident, 'bin/hab')} artifact upload #{::File.join(studio_path, '/src/results', artifact)}" }
+execute 'upload-pkg' do
+  command lazy { "#{hab_binary} pkg upload --url #{node['habitat-build']['depot-url']} --auth '#{depot_token}' #{::File.join(hab_studio_path, '/src/results', artifact)}" }
```

Comment:
Use of the `--auth` option is not ideal as it prints the token in the Delivery job console! Open to a better way here.